File: Android/app/src/main/java/com/diveapp/android/service/CommunityService.kt
package com.diveapp.android.service

import com.diveapp.android.model.CommunityCommentCreateRequest
import com.diveapp.android.model.CommunityCommentResponse
import com.diveapp.android.model.CommunityPostCreateRequest
import com.diveapp.android.model.CommunityPostListItem
import com.diveapp.android.model.CommunityPostResponse
import com.diveapp.android.model.CommunityPostUpdateRequest
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.DELETE
import retrofit2.http.GET
import retrofit2.http.PATCH
import retrofit2.http.POST
import retrofit2.http.Path
import retrofit2.http.Query

interface CommunityService {
    @POST("api/v1/community/posts")
    suspend fun createPost(@Body body: CommunityPostCreateRequest): CommunityPostResponse

    @GET("api/v1/community/posts")
    suspend fun listPosts(@Query("q") query: String? = null): List<CommunityPostListItem>

    @GET("api/v1/community/posts/mine")
    suspend fun listMyPosts(): List<CommunityPostListItem>

    @GET("api/v1/community/posts/{id}")
    suspend fun getPost(@Path("id") id: String): CommunityPostResponse

    @PATCH("api/v1/community/posts/{id}")
    suspend fun updatePost(@Path("id") id: String, @Body body: CommunityPostUpdateRequest): CommunityPostResponse

    @DELETE("api/v1/community/posts/{id}")
    suspend fun deletePost(@Path("id") id: String): Response<Unit>

    @POST("api/v1/community/posts/{id}/comments")
    suspend fun createComment(@Path("id") postId: String, @Body body: CommunityCommentCreateRequest): CommunityCommentResponse

    @GET("api/v1/community/posts/{id}/comments")
    suspend fun listComments(@Path("id") postId: String): List<CommunityCommentResponse>

    @DELETE("api/v1/community/comments/{id}")
    suspend fun deleteComment(@Path("id") id: String): Response<Unit>
}

File: Web/src/services/communityService.ts
import { apiClient } from '../core/network/apiClient';
import type {
  CommunityCommentResponse,
  CommunityPostCreateRequest,
  CommunityPostListItem,
  CommunityPostResponse,
} from '../models/communityModels';

export const communityService = {
  async createPost(body: CommunityPostCreateRequest): Promise<CommunityPostResponse> {
    const { data } = await apiClient.post<CommunityPostResponse>('/api/v1/community/posts', body);
    return data;
  },
  async listPosts(query?: string): Promise<CommunityPostListItem[]> {
    const { data } = await apiClient.get<CommunityPostListItem[]>('/api/v1/community/posts', {
      params: query ? { q: query } : undefined,
      noAuth: true,
    });
    return data;
  },
  async getPost(id: string): Promise<CommunityPostResponse> {
    const { data } = await apiClient.get<CommunityPostResponse>(`/api/v1/community/posts/${id}`, { noAuth: true });
    return data;
  },
  async updatePost(id: string, body: CommunityPostCreateRequest): Promise<CommunityPostResponse> {
    const { data } = await apiClient.patch<CommunityPostResponse>(`/api/v1/community/posts/${id}`, body);
    return data;
  },
  async deletePost(id: string): Promise<void> {
    await apiClient.delete(`/api/v1/community/posts/${id}`);
  },
  async createComment(postId: string, content: string): Promise<CommunityCommentResponse> {
    const { data } = await apiClient.post<CommunityCommentResponse>(`/api/v1/community/posts/${postId}/comments`, {
      content,
    });
    return data;
  },
  async listComments(postId: string): Promise<CommunityCommentResponse[]> {
    const { data } = await apiClient.get<CommunityCommentResponse[]>(`/api/v1/community/posts/${postId}/comments`, {
      noAuth: true,
    });
    return data;
  },
  async deleteComment(id: string): Promise<void> {
    await apiClient.delete(`/api/v1/community/comments/${id}`);
  },
};
